### PR TITLE
Chore/introduce linting

### LINKED ISSUE #985: Proposal: introducing linting to the kubescape codebase
**Is your feature request related to a problem? Please describe.**</br>
 > A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]
 
When a code base grows, code readability becomes paramount.
There are quite a few places where the kubescape codebase departs from widely adopted go practice.
We have all been there and this is perfectly understandable at this stage.
Pure style notwithstanding, this results in some components being harder to grasp than strictly needed.

**Describe the solution you'd like.**</br>
 > A clear and concise description of what you want to happen.

I propose to introduce linting to the kubescape git repository and gradually address issues.

* Existing linting issues should not block PRs unless they introduce offending code.
* Enabling linting rules is a gradual process: we'd start with a few rules only and gradually raise our enforcement of coding standards.
* We won't introduce rules that would break existing interfaces because of some styling issues (e.g. initialisms, etc.)
* Linting fixes are not always trivial. We should limit the scope of such fixes to a series of rather small, reviewable PRs.

**Describe alternatives you've considered.**</br>
 > A clear and concise description of any alternative solutions or features you've considered.
 
I've also tried to run some static analysis linters manually, detect some issues, and fix them individually.
This is the approach followed by #982.

**Additional context.**</br>
 > Add any other context or screenshots about the feature request here.

### PR DESCRIPTION
**Pull request by fredbi**
## Describe your changes
* Introduced minimal linters configuration, with a new github action to enforce linting rules.
* Fixed linting issues with this minimal config

NOTE: this PR also relates to #982, which already addresses a few linting issues (suspicious loops).

## This PR fixes:

* Resolved #985 

## Checklist before requesting a review

- [ ] My code follows the style guidelines of this project
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] New and existing unit tests pass locally with my changes

**Please open the PR against the `dev` branch (Unless the PR contains only documentation changes)**

### CODE REVIEW

**Comment by Turuk3100** [comment]
> <h3>Hi! Welcome to Kubescape. Thank you for taking the time and contributing to the open source community</h3><p><h4>We will try to review as soon as possible!</h4></p>

Нет войне!

File: core/pkg/opaprocessor/processorhandlerutils.go [hunk 1]
```diff
@@ -117,9 +117,11 @@ func getKubernetesObjects(k8sResources *cautils.K8SResources, allResources map[s
 					groupResources := k8sinterface.ResourceGroupToString(groups, version, resource)
 					for _, groupResource := range groupResources {
 						if k8sObj, ok := (*k8sResources)[groupResource]; ok {
-							if k8sObj == nil {
-								// logger.L().Debug("skipping", helpers.String("resource", groupResource))
-							}
+							/*
+								if k8sObj == nil {
```
Comment (dwertent) [inline core/pkg/opaprocessor/processorhandlerutils.go:121]: TODO: Delete this code

File: core/pkg/opaprocessor/processorhandler.go [hunk 2]
```diff
@@ -95,7 +98,7 @@ func (opap *OPAProcessor) Process(policies *cautils.Policies) error {
 
 	opap.loggerDoneScanning()
 
-	return errs
+	return nil
```
Comment (dwertent) [inline core/pkg/opaprocessor/processorhandler.go:101]: TODO: Reconsider returning a list of errors (as current behavior)

**Review by dwertent — APPROVED** [review]

File: core/pkg/opaprocessor/processorhandler.go (hunk 2 above)
Comment (fredbi) [reply core/pkg/opaprocessor/processorhandler.go:101]: @dwertent I agree, this is an interesting point.

The current behavior is to never actually hydrate "errs", hence the simplification. Indeed, the "Process" function never returns an error.

**Pull request merged**